processor/k8sprocessor/options.go

```diff
@@ -234,3 +234,17 @@ func WithFilterFields(filters ...FieldFilterConfig) Option {
 		return nil
 	}
 }
+
+func WithExtractPodAssociations(podAssociations ...PodAssociationConfig) Option {
+	return func(p *kubernetesprocessor) error {
+		associations := []kube.Association{}
```

Comment:
Preallocate?

```suggestion
		associations := make([]kube.Association, 0, len(podAssociations))
```

Btw why not directly act on `p.podAssociations.Associations`?